### DISCUSSION TITLE
Is Deletion Built into the User Input Objects Provided by Basalt?

### DISCUSSION THREAD

**Discussion by PaxiTheAlchemist** [Pyroxenium/Basalt#2] [Q&A] [answered]
# Question
Is deletion within the input and textField objects as seen in the preview gifs within the wiki built-in or created with logic outside of the basalt library? I am unsure if deletion was left out intentionally or by accident, but with nothing being talked about within the wiki on top of my testing so far, I have not been able to delete text within inputs or the textField objects without any implementation of deletion on my own and have assumed it was built-in from the preview gif. Is this assumption true?
## Other Notes
I have been doing all my testing so far on the Linux version of the CraftOS Emulator for ease of access. The current version of CraftOS I am using is 1.8 which its host is version 1.100.5 of ComputerCraft. Could this be a compatibility issue with the library and the emulator and deletion is in fact supported?

**Comment by NoryiE** [comment, marked as answer]
Hello, thanks. This was a small issue on my side. I was using the key number instead of just using keys.backspace. If you are downloading the new basalt version it should work (i tested it and it works fine for me).
Remember i am using now a minified version, if you want the "source"- file just download it from my source folder.

**Reply by PaxiTheAlchemist** [reply]
Thanks for the feedback! Removing the basalt I was using and running the auto-installer provided within the wiki has gotten me the correct version! Deletion using backspace now functions with my application!

**Reply by PaxiTheAlchemist** [reply]
Also forgot to mention that making a new line with the enter key did not work previously for the text field, but with this new basalt version, it now works too!

**Reply by NoryiE** [reply]
No problem! Deletion is of course a build in feature, also creating a new line is a default feature of textfields. You should also be able to use the arrow keys. Only the delete button is currently not available. I am improving textfield & input objects (make selection available), as soon as i am done with that i will also add the delete button.
